src/d3a/models/market/one_sided.py

```diff
@@ -49,15 +49,25 @@ def __init__(self, time_slot=None, bc=None, notification_listener=None,
         self.in_sim_duration = in_sim_duration
 
     def __repr__(self):  # pragma: no cover
-        return "<OneSidedMarket{} offers: {} (E: {} kWh V: {}) trades: {} (E: {} kWh, V: {})>"\
-            .format(" {}".format(self.time_slot_str),
-                    len(self.offers),
-                    sum(o.energy for o in self.offers.values()),
-                    sum(o.price for o in self.offers.values()),
-                    len(self.trades),
-                    self.accumulated_trade_energy,
-                    self.accumulated_trade_price
-                    )
+        return "<{}{} offers: {} (E: {} kWh V: {}) trades: {} (E: {} kWh, V: {})>"\
+            .format(
+                self._class_name,
+                " {}".format(self.time_slot_str),
+                len(self.offers),
+                sum(o.energy for o in self.offers.values()),
+                sum(o.price for o in self.offers.values()),
+                len(self.trades),
+                self.accumulated_trade_energy,
+                self.accumulated_trade_price
+            )
+
+    @property
+    def _class_name(self):
+        return "OneSidedMarket"
```

Comment:
Nit: this could also be `self.__class__.__name__` (in case we want to make it dynamic)

Reply:
Will do and also remove this from the classes that inherit from this